src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointPair.java

```diff
@@ -0,0 +1,476 @@
+     * Note: assuming input alignment regions are NOT completely enclosed in the other.
+     */
+    @VisibleForTesting
+    static BreakpointComplication resolveComplications(final ChimericAlignment chimericAlignment)
+            throws IOException {
+
+        final SimpleInterval leftReferenceInterval, rightReferenceInterval;
+        if(chimericAlignment.involvesReferenceIntervalSwitch) {
+            leftReferenceInterval  = chimericAlignment.regionWithHigherCoordOnContig.referenceInterval;
+            rightReferenceInterval = chimericAlignment.regionWithLowerCoordOnContig.referenceInterval;
+        } else  {
+            leftReferenceInterval  = chimericAlignment.regionWithLowerCoordOnContig.referenceInterval;
+            rightReferenceInterval = chimericAlignment.regionWithHigherCoordOnContig.referenceInterval;
+        }
+
+        final AlignmentRegion firstContigRegion  = chimericAlignment.regionWithLowerCoordOnContig,
+                              secondContigRegion = chimericAlignment.regionWithHigherCoordOnContig;
+
+        final int r1e = leftReferenceInterval.getEnd(),
+                  r2b = rightReferenceInterval.getStart(),
+                  c1e = chimericAlignment.regionWithLowerCoordOnContig.endInAssembledContig,
+                  c2b = chimericAlignment.regionWithHigherCoordOnContig.startInAssembledContig;
+
+        final byte[] contigSeq = chimericAlignment.contigSeq;
+
+        // a segment with lower coordinate on the locally-assembled contig could map to a higher reference coordinate region
+        // under two basic types of SV's: inversion (strand switch necessary) and translocation (no strand switch necessary)
+        if (chimericAlignment.strandSwitch!= ChimericAlignment.StrandSwitch.NO_SWITCH) { // the case involves an inversion
+            // TODO: 12/5/16 duplication detection to be done for inversion alleles
+            return new BreakpointComplication(getHomology(firstContigRegion, secondContigRegion, contigSeq), getInsertedSequence(firstContigRegion, secondContigRegion, contigSeq), EMPTY_STRING, 0, 0);
+        } else if (chimericAlignment.isNotSimpleTranslocation) { // no translocation
+
+            final int d1 = r2b - r1e - 1, // distance-1 between the two regions on reference
+                      d2 = c2b - c1e - 1; // distance-1 between the two regions on contig
+
+            String homologyForwardStrandRepresentation = EMPTY_STRING, insertedSeqForwardStrandRepresentation=EMPTY_STRING, dupSeqForwardStrandRepresentation=EMPTY_STRING;
+            int refDup=0, ctgDup=0;
+            // when d1<d2 (but not when both are negative), it always indicate an insertion, and vice versa
+            // when d2<0, there's always homologyForwardStrandRep
+            if (d1>0 && d2==0) {        // DEL: simple deletion, deleted sequence is [r1e+1, r2b-1] on the reference
+                homologyForwardStrandRepresentation    = EMPTY_STRING;
```

Comment:
The intent was to be more explicit in re-initializing (or rather assignment). Leaving them un-inilialized makes IDE unhappy.